tiledb/sm/filesystem/azure.cc

```diff
@@ -893,20 +898,25 @@ Status Azure::write_blocks(
     const URI& uri,
     const void* const buffer,
     const uint64_t length,
-    const bool /*last_block*/) {
+    const bool last_block) {
   if (!uri.is_azure()) {
     return LOG_STATUS(Status::AzureError(
         std::string("URI is not an Azure URI: " + uri.to_string())));
   }
 
-  // TODO: calculate number of parallel operations.
-  const uint64_t num_ops = 1;
+  // Ensure that each thread is responsible for exactly block_list_block_size_
```

Comment:
Do you mind to add a comment that "length is evenly divisible by block_list_block_size unless this is the last block" (this made more sense when I looked down at :916)

Reply:
Done -- added a similar comment in s3.cc